app/models/member.rb

```diff
@@ -45,6 +47,15 @@ def validate_namespace
     errors.add(namespace.type, 'namespace cannot have members')
   end
 
+  # Method to ensure we don't leave a group or project without an owner
+  def last_namespace_owner_member
```

Comment:
This only needs to apply to GroupMember, and it only has to be check last owner in the hierarchy. If you want this could be addressed in a separate PR.